paperscraper/pdf.py

```diff
@@ -157,8 +191,277 @@ def save_pdf_from_dump(
             logger.warning(f"Skipping {paper['title']} since no DOI available.")
             continue
         filename = paper[key_to_save].replace("/", "_")
+        pdf_file = Path(os.path.join(pdf_path, f"{filename}.pdf"))
+        xml_file = pdf_file.with_suffix(".xml")
+        if pdf_file.exists():
+            logger.info(f"File {pdf_file} already exists. Skipping download.")
+            continue
+        if xml_file.exists():
+            logger.info(f"File {xml_file} already exists. Skipping download.")
+            continue
+        output_path = str(pdf_file)
         save_pdf(
             paper,
-            os.path.join(pdf_path, f"{filename}.pdf"),
+            output_path,
             save_metadata=save_metadata,
+            api_keys=api_keys_dict
         )
+
+def load_api_keys(filepath):
+    """
+    Reads API keys from a file and returns them as a dictionary.
+    The file should have each API key on a separate line in the format:
+        KEY_NAME=API_KEY_VALUE
+
+    Example:
+        WILEY_TDM_API_TOKEN=your_wiley_token_here
+        ELSEVIER_TDM_API_KEY=your_elsevier_key_here
+
+    Args:
+        filepath (str): Path to the file containing API keys.
+
+    Returns:
+        Dict[str, str]: A dictionary where keys are API key names and values are their respective API keys.
+    """
+    api_keys = {}
+    try:
+        with open(filepath, "r") as f:
+            for line in f:
+                if "=" in line:
+                    key, value = line.strip().split("=", 1)
+                    api_keys[key] = value
+    except Exception as e:
+        print(f"Error reading API keys file: {e}")
+    return api_keys
+
+def fallback_wiley_api(paper_metadata: Dict[str, Any], output_path: Path, logger: logging.Logger, api_keys: Dict[str, str]):
```

Comment:
any reason why passing around a logger here, can we not use the standard logger? That would be more homogenous

Reply:
You're right, there is no good reason. I'll change it to the standard logger.